cns/restserver/api.go

```diff
@@ -1119,14 +1145,35 @@ func (service *HTTPRestService) publishNetworkContainer(w http.ResponseWriter, r
 	}
 
 	logger.Request(service.Name, &reqCopy, err)
+
+	// TODO - refactor this method for better error handling
 	if err != nil {
 		return
 	}
 
+	var ncParameters *cns.NetworkContainerParameters
+	ncParameters, err = getAuthTokenAndInterfaceIDFromNcURL(creteNcURLCopy)
+	if err != nil {
+		logger.Errorf("[Azure-CNS] nc parameters validation failed with %+v", err)
+		w.WriteHeader(http.StatusBadRequest)
+
+		badRequestResponse := &cns.PublishNetworkContainerResponse{
+			Response: cns.Response{
+				ReturnCode: http.StatusBadRequest,
+				Message:    fmt.Sprintf("Request contains a unexpected create url format in request body: %v", reqCopy.CreateNetworkContainerURL),
+			},
+			PublishErrorStr:   fmt.Sprintf("Bad request: Request contains a unexpected create url format in request body: %v", reqCopy.CreateNetworkContainerURL),
+			PublishStatusCode: http.StatusBadRequest,
+		}
+		err = service.Listener.Encode(w, &badRequestResponse)
+		logger.Response(service.Name, badRequestResponse, badRequestResponse.Response.ReturnCode, err)
+		return
+	}
+
 	switch r.Method {
 	case "POST":
 		// Join the network
-		publishResponse, publishError, err = service.joinNetwork(req.NetworkID, req.JoinNetworkURL)
+		publishResponse, publishError, err = service.joinNetwork(req.NetworkID)
```

Comment:
The existing code is __so__ bad that I am inclined to suggest just suppressing the linting error with
```
//nolint:bodyclose // please refactor this mess
```
since blindly appeasing the linter doesn't address real issues. Why is this code bad? Because
- `joinNetwork` has __already__ closed the request body, if there was one to close, even though it consumed __none__ of the body bytes. This causes inefficient TCP connection reuse, not sure what the effect is since it's agains the Wireserver host.
- The `publishResponse` var is reassigned below, with the result of a different http request. `defer`ing a body close earlier in the function __may__ have the equivalent effect while addressing the lint error, but it just really obfuscates that we are really handling 2 different `io.Readers`
- Control flow and scope of variables is puzzling.

Reply:
Cool, yeah I put a comment on there